core/app/models/spree/product.rb

```diff
@@ -105,7 +105,7 @@ class Product < Spree::Base
       validates :price, if: proc { Spree::Config[:require_master_price] }
     end
 
-    validates :slug, length: { minimum: 3 }, allow_blank: true, uniqueness: true
+    validates :slug, allow_blank: true, uniqueness: true
```

Comment:
@krzysiek1507 I realized there was a CI failure and, on further investigation, discovered that there was a spec indeed (not sure why it was an integration spec). It was checking that the slug cannot be too short. And, since this is the removal of functionality, I think the correct way to go about specs is to remove the code and specs and ensure that all the other specs are passing.